### ISSUE TITLE
Module creation of Integration does not work

### ISSUE THREAD

**Issue by jpreed2** [magento/magento2#4023]
## Steps to reproduce
1. Followed the instructions here http://devdocs.magento.com/guides/v2.0/howdoi/webapi/integration.html
## Expected result
1. After installing and enabling the module the System->Extensions->Integrations lists the new Integration generated by the module.
## Actual result
1. New module shows as enabled.
2. New Integration does not get created and is not listed in the System->Extensions->Integrations list.

Either something is missing in the documentation or this functionality is not working.

**Comment by KrystynaKabannyk** [comment]
The internal ticket is created MAGETWO-51642.

**Comment by keharper** [comment]
The Integration article has been updated. It was missing a step about creating an install class.

**Comment by jpreed2** [comment]
I have followed the instructions and still can't get it to work.  It never executes the InstallData.php.  I've tried removing the entire directory and running "magento setup:upgrade" and I see the module is removed from the list.  I then try adding it back and running "magento setup:upgrade" again and it shows the module as installed.  But the setup is not getting triggered.

I put some error_log() statements in the constructor of InstallData.php to see if it's getting instantiated, and it's not.

I'm using 2.0.4.  Is there a trick to get it to execute the InstallData.php after it had been previously installed?  Or does it only run once and never again?

Thanks.

**Comment by jpreed2** [comment]
BTW, providing a sample module zip file in the documentation that's confirmed working would be very helpful.  All it takes is one wrong lower or upper case letter to prevent things from working, which is hard to detect without a sample.

**Comment by jpreed2** [comment]
Ah, finally I figured it out!  I had to manually remove the module from the setup_module table.

delete from magento.setup_module where module = 'module_name';

Then I ran "magento setup:upgrade" and it worked!

Thanks.

**Comment by jpreed2** [comment]
Where is there a list of all the available API modules to specify in the api.xml for access?

Examples:

```
            <resource name="Magento_Customer::customer" />
            <resource name="Magento_Log::online" />
```

Is there a way to specify access to all modules?

**Comment by jpreed2** [comment]
OK, answered my own questions again.  Found the access levels are listed in the authorization_rule table:

select \* from authorization_rule;

And the authorization for all is:

```
<resource name="Magento_Backend::all" />
```

**Comment by keharper** [comment]
I'm glad you were able to work out the problems. 

Did you have to manually delete the module from the `setup_module` table because you had unsuccessful attempts to create it?

I'll look into what our policy is about providing a ZIP file on devdocs. 

I'll add the the `<resource name="Magento_Backend::all" />` statement to the doc.

**Comment by jpreed2** [comment]
Yes, it appears once you install a module it will never run the InstallData.php again.  Since I had installed the module previously without the InstallData.php script it was ignoring it.

I'm guessing if you uninstall it and install again it would work, but there is no way to uninstall a module unless it was installed via composer.  So I had to manually remove it from the setup_module table.  This forces setup:upgrade to run the InstallData.php again.

Thanks.
